dbsc/workflow/rules/freesurfer.smk

```diff
@@ -1,113 +1,91 @@
-from os.path import join 
+from pathlib import Path
+from functools import partial
+
+# Directories
+freesurfer_dir = str(Path(config["output_dir"]) / "freesurfer")
+
+# Make directory if it doesn't exist
+Path(freesurfer_dir).mkdir(parents=True, exist_ok=True)
+
+# BIDS partials
+bids_fs_out = partial(
+    bids,
+    root=freesurfer_dir,
+    datatype="anat",
+    **config["subj_wildcards"],
+)
+
+# Freesurfer references (with additional in rules as necessary)
+# B. Fischl, A. van der Kouwe, C. Destrieux, E. Halgren, F. Ségonne, D.H. Salat, E. Busa, L.J. Seidman, J. Goldstein, D. Kennedy, V. Caviness, N. Makris, B. Rosen, A.M. Dale. Automatically parcellating the human cerebral cortex. Cereb. Cortex, 14 (2004), pp. 11-22, 10.1093/cercor/bhg087
+
 
 rule thalamic_segmentation:
+    """Perform thalamus segmentation
+
+    Reference: J.E. Iglesias, R. Insausti, G. Lerma-Usabiaga, M. Bocchetta, K. Van Leemput, D.N. Greve, A. van der Kouwe, B. Fischl, C. Caballero-Gaudes, P.M. Paz-Alonso. A probabilistic atlas of the human thalamic nuclei combining ex vivo MRI and histology. NeuroImage, 183 (2018), pp. 314-326, 10.1016/j.neuroimage.2018.08.012
+
+    NOTE: Output file is defined by Freesurfer script
+    """
     input:
-        freesurfer_dir=join(config['bids_dir'], 'derivatives/freesurfer'),
+        freesurfer_dir=freesurfer_dir,
     params:
         fs_license=config["fs_license"],
-        threads=workflow.cores,
     output:
-        thal_seg=bids(
-            root=join(config["output_dir"], "freesurfer"),
-            datatype="anat",
-            suffix="T1.mgz"
-            space="Freesurfer",
-            **config["subj_wildcards"],
-        ),
-    container: 
-        config['singularity']['freesurfer']
+        thal_seg=str(Path(freesurfer_dir) / "{subject}/mri/ThalamicNuclei.v12.T1.mgz"),
+    threads: workflow.cores
+    container:
+        config["singularity"]["freesurfer"]
     shell:
-        "FS_LICENSE={params.fs_license} "
-        "ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS={params.threads} "
-        "SUBJECTS_DIR={input.freesurfer_dir} "
-        "segmentThalamicNuclei.sh {{subject}}"
+        "FS_LICENSE={params.fs_license} && "
+        "ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS={threads} && "
+        "SUBJECTS_DIR={input.freesurfer_dir} && "
+        "segmentThalamicNuclei.sh {wildcards.subject}"
+
 
-rule mgz_to_nii:
-    input: 
+rule mgz2nii:
+    """
+    Convert from .mgz to .nii.gz
+
+    NOTE: During conversion, files are renamed to BIDS-esque formatting
+    """
+    input:
         thal=rules.thalamic_segmentation.output.thal_seg,
-        aparcaseg=join(config["output_dir"], "freesurfer/{subject}/mri/aparc+aseg.mgz")
-        lRibbon=join(config["output_dir"], "freesurfer/{subject}/mri/lh.ribbon.mgz")
-        rRibbon=join(config["output_dir"], "freesurfer/{subject}/mri/rh.ribbon.mgz")
+        aparcaseg=str(Path(freesurfer_dir) / "sub-{subject}/mri/aparc+aseg.mgz"),
```

Comment:
These files should already exist since they're created by freesurfer itself which is a prereq to this workflow at the moment.

Reply:
Gotcha. In that case, I think we should get rid of the directory creation command on Line 8 in this workflow, since it definitely won't work if the directory doesn't exist.